js/main.js

```diff
@@ -1,2 +1,29 @@
 /* global data */
 /* exported data */
+var $form = document.querySelector('form');
+$form.addEventListener('submit', submitButton);
+
+var $photo = document.querySelector('img');
+var $photoURL = document.querySelector('#photoURL');
+$photoURL.addEventListener('input', updateURL);
+
+function updateURL(event) {
+  var src = $photoURL.value;
+  $photo.setAttribute('src', src);
+}
+
+function submitButton(event) {
+  event.preventDefault();
+  data.nextEntryId += 1;
+  var inputObj = {
+    title: $form.elements.title.value,
+    imageURL: $form.elements.photoURL.value,
+    notes: $form.elements.notes.value,
+    nextEntryId: data.nextEntryId
+  };
+  data.entries.unshift(inputObj);
+  var inputJSON = JSON.stringify(data);
+  localStorage.setItem('Input Form', inputJSON);
```

Comment:
This shouldn't happen every time the form is submitted, you only need to update localStorage right before the page unloads. I suggest looking back at `javascript-local-storage` for a refresher on how this can be accomplished.